packages/frontend-2/components/project/webhooks-page/DeleteWebhookDialog.vue

```diff
@@ -0,0 +1,128 @@
+<template>
+  <LayoutDialog
+    v-model:open="isOpen"
+    max-width="sm"
+    title="Delete Webhook"
+    :buttons="dialogButtons"
+  >
+    <div class="flex flex-col gap-6 text-sm text-foreground">
+      <p>
+        Are you sure you want to
+        <strong>permanently delete</strong>
+        the selected webhook?
+      </p>
+      <div v-if="webhook" class="flex flex-col gap-2">
+        <strong>{{ webhook.description }}</strong>
+      </div>
+
+      <p>
+        This
+        <strong>cannot</strong>
+        be undone.
+      </p>
+    </div>
+  </LayoutDialog>
+</template>
+
+<script setup lang="ts">
+import { useMutation } from '@vue/apollo-composable'
+import { LayoutDialog, TableItemType } from '@speckle/ui-components'
+import { WebhookItem } from '~~/lib/projects/helpers/types'
+import { deleteWebhookMutation } from '~~/lib/projects/graphql/mutations'
+import {
+  ROOT_QUERY,
+  convertThrowIntoFetchResult,
+  getCacheId,
+  getFirstErrorMessage,
+  modifyObjectFields
+} from '~~/lib/common/helpers/graphql'
+import { useGlobalToast, ToastNotificationType } from '~~/lib/common/composables/toast'
+import { WebhookCollection } from '~~/lib/common/generated/gql/graphql'
+
+const props = defineProps<{
+  open: boolean
+  webhook: TableItemType<WebhookItem> | null
+}>()
+
+const { triggerNotification } = useGlobalToast()
+const { mutate: deleteMutation } = useMutation(deleteWebhookMutation)
+
+const isOpen = defineModel<boolean>('open', { required: true })
+
+const deleteConfirmed = async () => {
+  const webhookId = props.webhook?.id
+  if (!webhookId) {
+    return
+  }
+
+  const result = await deleteMutation(
+    {
+      webhook: {
+        id: props.webhook.id,
+        streamId: props.webhook.streamId
+      }
+    },
+    {
+      update: (cache, { data }) => {
+        if (data?.webhookDelete) {
+          const cacheId = getCacheId('Webhook', webhookId)
+          cache.evict({
+            id: cacheId
+          })
+
+          modifyObjectFields<undefined, { [key: string]: WebhookCollection }>(
```

Comment:
There's no way this could ever work, because `ROOT_QUERY` has no such `webhooks` field. I suggest always validating cache mutators with a debugger to ensure they actually work.

Considering that we want to update `Project.webhooks` and `Project` can be uniquely identified (has an `id` field)_, there's no reason for us to start at the ROOT_QUERY level. You can invoke `modifyObjectFields` on `getCacheId('Project', projectId)`.